src/c.c

```diff
@@ -26,13 +26,33 @@ SEXP vctrs_c(SEXP call, SEXP op, SEXP args, SEXP env) {
   return out;
 }
 
+static inline bool needs_vec_c_fallback(SEXP xs);
+static SEXP vec_c_fallback(SEXP xs);
+
 // [[ include("vctrs.h") ]]
 SEXP vec_c(SEXP xs,
            SEXP ptype,
            SEXP name_spec,
            enum name_repair_arg name_repair) {
   R_len_t n = Rf_length(xs);
 
+  if (needs_vec_c_fallback(xs)) {
```

Comment:
Should the `ptype` be considered here? What if I did `vec_c(<Period>, <Period>, .ptype = logical())`?

Reply:
Good point. Probably best to give the same kind of error as with `name_spec`.